test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -0,0 +1,561 @@
+            realm.query(Sample::class, "stringField = 42")
+        }.let {
+            assertTrue(it.message!!.contains("Wrong query field"))
+        }
+
+        assertFailsWith<IllegalArgumentException> {
+            realm.query(Sample::class, "nonExistingField = 13")
+        }.let {
+            assertTrue(it.message!!.contains("Wrong query field"))
+        }
+    }
+
+    @Test
+    @Ignore
+    fun asFlow() {
+        // TODO
+    }
+
+    @Test
+    @Ignore
+    fun asFlow_throwsInsideWrite() {
+        // TODO
+    }
+
+    @Test
+    fun composedQuery() {
+        val value1 = 1
+        val value2 = 2
+        val value3 = 3
+        val value4 = 4
+
+        realm.writeBlocking {
+            copyToRealm(Sample().apply { intField = value1 })
+            copyToRealm(Sample().apply { intField = value2 })
+            copyToRealm(Sample().apply { intField = value3 })
+            copyToRealm(Sample().apply { intField = value4 })
+        }
+
+        realm.query(Sample::class, "intField > 1")
+            .query("intField > 2")
+            .find()
+            .let { results ->
+                assertEquals(2, results.size)
+            }
+    }
+
+    @Test
+    fun composedQuery_withDescriptor() {
+        val value1 = 1
+        val value2 = 2
+        val value3 = 3
+        val value4 = 4
+
+        realm.writeBlocking {
+            copyToRealm(Sample().apply { intField = value1 })
+            copyToRealm(Sample().apply { intField = value2 })
+            copyToRealm(Sample().apply { intField = value3 })
+            copyToRealm(Sample().apply { intField = value4 })
+        }
+
+        realm.query(Sample::class, "intField > 1")
+            .query("intField > 2")
+            .sort(Sample::intField.name, QuerySort.DESCENDING)
+            .find()
+            .let { results ->
+                assertEquals(2, results.size)
+                assertEquals(value4, results[0].intField)
+                assertEquals(value3, results[1].intField)
+            }
+
+        realm.query(Sample::class, "intField > 1")
+            .query("intField > 2")
+            .sort(Sample::intField.name, QuerySort.DESCENDING)
+            .limit(1)
+            .find()
+            .let { results ->
+                assertEquals(1, results.size)
+                assertEquals(value4, results[0].intField)
+            }
+    }
+
+    @Test
+    fun composedQuery_withDescriptorAndQueryAgain() {
+        val value1 = 1
+        val value2 = 2
+        val value3 = 3
+        val value4 = 4
+
+        realm.writeBlocking {
+            copyToRealm(Sample().apply { intField = value1 })
+            copyToRealm(Sample().apply { intField = value2 })
+            copyToRealm(Sample().apply { intField = value3 })
+            copyToRealm(Sample().apply { intField = value4 })
+        }
+
+        realm.query(Sample::class, "intField > 1")
+            .query("intField > 2")
+            .sort(Sample::intField.name, QuerySort.DESCENDING)
```

Comment:
Not 100% sure if this test is interesting...the sort doesn't do anything here... Maybe if you used limit instead you could write a test where this actually had an impact.

Reply:
Generally, if this test was for testing query -> descriptor -> query, we should cover all the descriptors in the test.